### PR TITLE
Only report missing STS header on domain redirects at Low threshold.

### PR DESCRIPTION
**Pull request by psiinon**
Justification: https://twitter.com/psiinon/status/1080782442259234817
Might be worth seeing if there are other responses later today before merging?
Can of course discuss any concern here too ;)

### CODE REVIEW

File: src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java [hunk 1]
```diff
@@ -104,7 +107,25 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 		if (msg.getRequestHeader().isSecure()){ //No point reporting missing for non-SSL resources
 			//Content available via both HTTPS and HTTP is a separate though related issue
 			if (stsOption == null) { // Header NOT found
+				boolean report = true;
+				if (! this.getAlertThreshold().equals(AlertThreshold.LOW) &&
+						HttpStatusCode.isRedirection(msg.getResponseHeader().getStatusCode())) {
+					// Only only report https redirects to the same domain at low threshold
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java]: Duplicated word.

File: src/org/zaproxy/zap/extension/pscanrulesAlpha/ZapAddOn.xml [hunk 2]
```diff
@@ -10,8 +10,9 @@
 	Fix a stack overflow with PII Scanner.<br>
 	Fix a DateParseException in Cacheable Scanner (Issue 4969).<br>
 	Fix Open Redirect (10028) "Attack" should be Desc.<br>
-        Fix false positive due to RxJS Observable method being mistaken for ASP source disclosure.<br>
+	Fix false positive due to RxJS Observable method being mistaken for ASP source disclosure.<br>
 	Tweak User Controllable Charset and Cookie Poisoning to use Description/Other Info field instead of Attack (Issue 5149).<br>
+	Only report missing STS header on redirects to HTTPS URLs on the same domain at Low threshold.
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrulesAlpha/ZapAddOn.xml]: `<br>` at the end.

File: test/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScannerUnitTest.java [hunk 3]
```diff
@@ -201,4 +202,72 @@ public void shouldRaiseAlertIfResponseContainsStsHeaderAndMeta() throws URIExcep
...
+		msg.getResponseHeader().setStatusCode(301);
+		msg.getResponseHeader().addHeader(HttpHeader.LOCATION, "/default/");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertThat(alertsRaised.size(), equalTo(0));
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfThresholdLowRedirectRelativePath() throws URIException {
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScannerUnitTest.java]: `shouldRaise`?

**Comment by psiinon** [comment]
Fixed those issues.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**